src/app/components/project-map/project-map.component.ts

```diff
@@ -386,6 +390,30 @@ export class ProjectMapComponent implements OnInit, OnDestroy {
   resetZoom() {
     this.mapScaleService.resetToDefault();
   }
+
+  importProject() {
+    let uuid: string = '';
+    const dialogRef = this.dialog.open(ImportProjectDialogComponent, {
+      width: '400px',
+      autoFocus: false
+    });
+    let instance = dialogRef.componentInstance;
+    instance.server = this.server;
+    const subscription = dialogRef.componentInstance.onImportProject.subscribe((projectId: string) => {
+      uuid = projectId;
+    });
+
+    dialogRef.afterClosed().subscribe(() => {
+      subscription.unsubscribe();
+      this.projectService.open(this.server, uuid).subscribe(() => {
+        this.router.navigate(['/server', this.server.id, 'project', uuid]);
```

Comment:
@ziajka In my opinion we should redirect to imported project when calling from map page, i'm not sure what about project page

Reply:
In my option we should also redirect in case of new project. @piotrpekala7